src/Utilities/DoubleDicts.jl

```diff
@@ -160,42 +176,34 @@ function Base.getindex(
     d::AbstractDoubleDict,
     key::MOI.ConstraintIndex{F,S},
 ) where {F,S}
-    inner = getindex(d, F, S)
+    inner = d[F, S]
     return inner[key]
 end
 
-function Base.getindex(d::DoubleDict, ::Type{F}, ::Type{S}) where {F,S}
-    return DoubleDictInner{F,S}(d)
-end
-
 function Base.getindex(
     d::AbstractDoubleDictInner{F,S},
     key::MOI.ConstraintIndex{F,S},
 ) where {F,S}
-    if !haskey(d.inner, key.value)
+    if !haskey(d, key)
```

Comment:
get? with a default to nothing?

Reply:
The value might be `nothing`. And we need to throw `key` as the error, not `key.value`.

Reply:
then
```julia
function Base.getindex(
    d::AbstractDoubleDictInner{F,S},
    key::MOI.ConstraintIndex{F,S},
) where {F,S}
    x = get(d.inner, key.value) do
        throw(KeyError(key))
    end
    return typed_value(d, x)
end
```

Reply:
Oooo. I didn't consider this pattern! Nice.